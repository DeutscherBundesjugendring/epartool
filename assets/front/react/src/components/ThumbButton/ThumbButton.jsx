import React from 'react';
import PropTypes from 'prop-types';

const ThumbButton = ({
  disabled,
  label,
  onClick,
  type,
}) => {
  const linkClasses = 'btn btn-default btn-xs';
  let iconClasses = 'glyphicon ';

  if (type === 'like') {
    iconClasses += ' glyphicon-thumbs-up';
  } else if (type === 'dislike') {
    iconClasses += ' glyphicon-thumbs-down';
  }

  return (
    <button
      type="button"
      onClick={(e) => {
        e.stopPropagation();
        if (!disabled) {
          onClick();
        }
      }}
      className={linkClasses}
      disabled={disabled}
    >
      <span className="text-center offset-top-small offset-right-small small">
        {label}
      </span>
      <span className={iconClasses} aria-hidden="true" />
    </button>
  );
};

ThumbButton.defaultProps = {
  disabled: false,
};

ThumbButton.propTypes = {
  disabled: PropTypes.bool,
  label: PropTypes.string.isRequired,
  onClick: PropTypes.func.isRequired,
  type: PropTypes.oneOf(['like', 'dislike']).isRequired,
};

export default ThumbButton;
