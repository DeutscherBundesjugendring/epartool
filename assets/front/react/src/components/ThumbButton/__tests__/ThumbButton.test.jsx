import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import sinon from 'sinon';
import React from 'react';
import injectTapEventPlugin from 'react-tap-event-plugin';
import ThumbButton from '../ThumbButton';


injectTapEventPlugin();

describe('rendering', () => {
  it('renders correctly like', () => {
    const tree = shallow(
      <ThumbButton type="like" onTouchTap={() => {}} label="label" />
    );

    expect(shallowToJson(tree)).toMatchSnapshot();
  });

  it('renders correctly dislike', () => {
    const tree = shallow(
      <ThumbButton type="dislike" onTouchTap={() => {}} label="label" />
    );

    expect(shallowToJson(tree)).toMatchSnapshot();
  });

  it('renders correctly disabled like', () => {
    const tree = shallow(
      <ThumbButton type="like" onTouchTap={() => {}} disabled label="label" />
    );

    expect(shallowToJson(tree)).toMatchSnapshot();
  });

  it('renders correctly disabled dislike', () => {
    const tree = shallow(
      <ThumbButton type="dislike" onTouchTap={() => {}} disabled label="label" />
    );

    expect(shallowToJson(tree)).toMatchSnapshot();
  });
});

describe('functionality', () => {
  it('calls onTouchTap', () => {
    const spy = sinon.spy();
    const component = shallow(
      <ThumbButton type="like" onTouchTap={spy} label="label" />
    );

    component.simulate('touchTap', { stopPropagation: () => {} });
    expect(spy.calledOnce).toEqual(true);
  });
});
