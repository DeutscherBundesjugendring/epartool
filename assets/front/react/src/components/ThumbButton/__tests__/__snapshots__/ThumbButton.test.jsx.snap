// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering renders correctly disabled dislike 1`] = `
<button
  className="btn btn-default btn-xs"
  disabled={true}
  onTouchTap={[Function]}
>
  <span
    className="text-center offset-top-small offset-right-small small"
  >
    label
  </span>
  <span
    aria-hidden="true"
    className="glyphicon  glyphicon-thumbs-down"
  />
</button>
`;

exports[`rendering renders correctly disabled like 1`] = `
<button
  className="btn btn-default btn-xs"
  disabled={true}
  onTouchTap={[Function]}
>
  <span
    className="text-center offset-top-small offset-right-small small"
  >
    label
  </span>
  <span
    aria-hidden="true"
    className="glyphicon  glyphicon-thumbs-up"
  />
</button>
`;

exports[`rendering renders correctly dislike 1`] = `
<button
  className="btn btn-default btn-xs"
  disabled={false}
  onTouchTap={[Function]}
>
  <span
    className="text-center offset-top-small offset-right-small small"
  >
    label
  </span>
  <span
    aria-hidden="true"
    className="glyphicon  glyphicon-thumbs-down"
  />
</button>
`;

exports[`rendering renders correctly like 1`] = `
<button
  className="btn btn-default btn-xs"
  disabled={false}
  onTouchTap={[Function]}
>
  <span
    className="text-center offset-top-small offset-right-small small"
  >
    label
  </span>
  <span
    aria-hidden="true"
    className="glyphicon  glyphicon-thumbs-up"
  />
</button>
`;
